### PR TITLE
quadcontrol: read PID configuration from /etc/quad.conf

### PR DESCRIPTION
**Pull request by HBuczynski**
## Description

PP-24

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: quadcontrol/control.c [hunk 1]
```diff
@@ -190,6 +191,91 @@ static int quad_run(void)
...
+		var = line;
+		strVal = strchr(line, '=');
+		if (strVal == NULL) {
+			fprintf(stderr, "quad-control: wrong line %s in file %s\n", line, PATH_PIDS_CONFIG);
+			err = -EINVAL;
+			break;
+		}
+
+		*strVal++ = '\0';
+		val = strtof(strVal, NULL);
```
Comment (MaciejPurski) [inline quadcontrol/control.c:264]: You could use the second parameter to `strtof()` and check if the conversion was successful.

File: quadcontrol/control.c [hunk 2]
```diff
@@ -190,6 +191,91 @@ static int quad_run(void)
...
+
+		if (quad_parsePid(i - 1, var, val) < 0) {
```
Comment (MaciejPurski) [inline quadcontrol/control.c:266]: Is it possible that `i` is equal to 0 at this point? If yes, you would pass `-1`, which would then be interpreted as `UINT_MAX`.

File: quadcontrol/control.c [hunk 3]
```diff
@@ -190,6 +191,91 @@ static int quad_run(void)
...
+
+static int quad_readConfig(void)
+{
+	float val;
+	int err = EOK;
+	unsigned int i = 0;
+	size_t len = 0;
+	char *line = NULL, *var, *strVal;
+
+	quad_common.conf = fopen(PATH_PIDS_CONFIG, "r");
```
Comment (MaciejPurski) [inline quadcontrol/control.c:239]: You never close this file.

**Review by MaciejPurski — CHANGES_REQUESTED** [review]

File: quadcontrol/control.c (hunk 3 above)
Comment (HBuczynski) [reply quadcontrol/control.c]: Thanks, fixed

File: quadcontrol/control.c (hunk 2 above)
Comment (HBuczynski) [reply quadcontrol/control.c:266]: Yes it is. 
However, `quad_parsePid` handles this case.

Comment (MaciejPurski) [reply quadcontrol/control.c:266]: Ok, not the prettiest solution, but since it works as expected, then let's leave it like it is.

**Review by MaciejPurski — APPROVED** [review]

**Review by MaciejPurski — APPROVED** [review]

**Pull request merged**